src/types.rs

```diff
@@ -350,7 +353,12 @@ pub(crate) struct DirectoryUrls {
     pub(crate) new_nonce: String,
     pub(crate) new_account: String,
     pub(crate) new_order: String,
-    pub(crate) revoke_cert: String,
+    // The fields below were added later and old `AccountCredentials` may not have it.
+    // Newer deserialized account credentials grab a fresh set of `DirectoryUrls` on
+    // deserialization, so they should be fine. Newer fields should be optional, too.
+    pub(crate) new_authz: Option<String>,
+    pub(crate) revoke_cert: Option<String>,
+    pub(crate) key_change: Option<String>,
```

Comment:
I actually don't want to update `old_credentials`, because that reflects the old usage that I'm trying to still support to the extent possible. From that perspective `new_credentials` is testing the "new old" behavior, so probably also want to keep it as is. Can't think of another one for which adding it would have much value?